algorithms/CPlusPlus/Maths/power-of-two.cpp

```diff
@@ -1,40 +1,50 @@
 #include <iostream>
 using namespace std;
 
-//function starts
-bool isPowerOfTwo(const n)
+// function starts
+bool isPowerOfTwo(const int n)
 {
-    //declare a variable to know if n is a power of 2 or not
+    // declare a variable to know if n is a power of 2 or not
     long i = 1;
-    //at every iteration it will calcalute power of 2 starting from 1
+    // at every iteration it will calcalute power of 2 starting from 1
     while (i < n)
     {
         i = i * 2;
     }
-    //if n is the power of 2, i and n value will be same
-    //if they are same, it will return true, else it will return false
+    // if n is the power of 2, i and n value will be same
+    // if they are same, it will return true, else it will return false
     return i == n;
 }
 
-//main starts
+// Binary Method to check number is in power of two or not.
+// Time Complexity of this method is O(1)
+bool ispoweroftwo(const int n)
+{
+    // if number is in power of two it will give 0, otherwise it will not zero.
+    return (n & (n - 1)) == 0;
```

Comment:
what do you think the result will be if you give n = 0 for this function?

Reply:
Yes, I forgot this case sorry i will add this case.

Reply:
@Ashborn-SM  change code. kindly review it.